### PR TITLE
Add doctest for `IdT`

### PR DESCRIPTION
**Pull request by bulatkhabib**
PR for part of #2479

Adding doctest for IdT typeclass.

### CODE REVIEW

File: core/src/main/scala/cats/data/IdT.scala [hunk 1]
```diff
@@ -27,24 +27,86 @@ package data
...
 
   /**
    * Modify the context `F` using transformation `f`.
+   *
+   * Example:
+   * {{{
+   *  scala> import cats.~>
+   *  scala> import cats.data.IdT
+   *
+   *  scala> val vectorToList: Vector ~> List = new ~>[Vector, List] { override def apply[A](v: Vector[A]): List[A] = v.toList }
```
Comment (satorg) [inline core/src/main/scala/cats/data/IdT.scala:51]: This part can be simplified :
```suggestion
   *  scala> val vectorToList: Vector ~> List = FunctionK.liftFunction(_.toList)
```
(requires `import cats.arrow.FunctionK`)

Comment (satorg) [reply core/src/main/scala/cats/data/IdT.scala:51]: (for Scala3 the syntax is going to be slightly different though)

Comment (satorg) [reply core/src/main/scala/cats/data/IdT.scala:51]: Well, I'm leaning to think that examples in scaladocs should keep the examples as simple as possible. Whereas the details on how, let's say, `FunctionK` works is supposed to be captured in the `FunctionK` scaladocs per se.

But I don't think it's really worth bikeshedding anyway – in any case it is a great job on providing the missing docs for the class, thank you for doing this!

**Review by satorg — APPROVED** [review]

File: core/src/main/scala/cats/data/IdT.scala (hunk 1 above)
Comment (armanbilge) [reply core/src/main/scala/cats/data/IdT.scala:51]: I'm not sure if we can write it like that, don't the doctests have to cross-compile for Scala 2 and Scala 3? So we are stuck with worst-of-both-worlds syntax.

**Review by armanbilge — APPROVED** [review]
Just a couple nits, but overall this looks great and I think we should get it into the next release. Thanks for all your work, and sorry for the slow review cycle!

**Pull request merged**